sklearn/ensemble/gradient_boosting.py

```diff
@@ -1846,3 +1853,467 @@ def apply(self, X):
+                       for params in param_iter)
+    
+        n_splits = int(len(out)/len(param_iter))
+    
+        grid_scores = []
+        i = 0
+        best_mean = -np.inf
+        for params in param_iter:
+        
+            score_list = []
+            n_est_list = []
+            for idx in range(n_splits):
+                score, n_estimators = out[i]
+                n_est_list.append(n_estimators)
+                score_list.append(score)
+                i += 1
+        
+            scores = np.array(score_list)
+            mean = np.mean(scores)
+            params['n_estimators'] = int(np.mean(n_est_list))
+            score_tuple = _CVScoreTuple(params, mean, scores)
+            grid_scores.append(_CVScoreTuple(params, mean, scores))
+        
+            if mean > best_mean:
+                best_tuple = score_tuple
+                best_params = params
+        
+        self.grid_scores_ = grid_scores
+        self.best_score_tuple_ = best_tuple
+        self.best_params_ = best_params
+        
+        if self.refit:
+            self.best_params_['random_state'] = self.random_state
+            gb = self._estimator_class(**self.best_params_)
+            gb.fit(X, y)
+            self.best_estimator_ = gb
+    
+        return self
+    
+    def predict(self, X):
+        """Call `predict` on the best estimator.
+    
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+        Returns
+        -------
+        y: array of shape = ["n_samples]
+            The predicted values.
+        """
+        return self.best_estimator_.predict(X)
+
+    def predict_proba(self, X):
+        """Call `predict_proba` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+
+        Raises
+        ------
+        AttributeError
+            If the ``loss`` does not support probabilities.
+
+        Returns
+        -------
+        p : array of shape = [n_samples]
+            The class probabilities of the input samples. The order of the
+            classes corresponds to that in the attribute `classes_`.
+        """
+        return self.best_estimator_.predict_proba(X)
+
+    def predict_log_proba(self, X):
+        """Call `predict_log_proba` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+        Raises
+        ------
+
+        AttributeError
+            If the ``loss`` does not support probabilities.
+
+        Returns
+        -------
+        p : array of shape = [n_samples]
+            The class probabilities of the input samples. The order of the
+            classes corresponds to that in the attribute `classes_`.
+        """
+        return self.best_estimator_.predict_log_proba(X)
```

Comment:
You can inherit from `predict`, `predict_proba` and `predict_log_proba` from `GridSearchCV` right?